android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt

```diff
@@ -114,6 +114,27 @@ class RNCIOPushMessaging(
         }
     }
 
+    /**
+     * Get the registered device token for the app.
+     * @returns Promise with device token as a string, or error if no token is
+     * registered or the method fails to fetch token.
+    */
+    @ReactMethod
+    fun getRegisteredDeviceToken(promise: Promise) {
+        try {
+            // Get the device token from SDK
+            val deviceToken: String? = CustomerIO.instance().registeredDeviceToken
+
+            if (deviceToken != null) {
+                promise.resolve(deviceToken)
+            } else {
+                promise.reject("device_token_not_found", "The device token is not available.")
+            }
```

Comment:
I would have preferred to have the Promise resolve null instead of reject when the native device token is null. 

A device not yet having a device token is not an error, IMO.